### PR TITLE
Allow pipeline to automatically create pull requests with Github CLI 

### PR DESCRIPTION
**Pull request by Jolisa-Brown**
🛠️ Description
These changes in the publish-api yaml file limit the pipeline's dispatch workflow to use exclusively the Github cli for autogenerating pull requests instead of a 3rd party action. This change minimizes platform exposure upon a 3rd party action's deprecation or improper maintenance.

📒 Additional Notes
This update is being executed in all services with write permissions for creating PRs. The three repos inclusive of these privileges are cloud-api, cloud-ui, and hcp-sdk-go as reflected in the PRs linked below.

❓ Why of the Changes
The security team has decided the platform should prioritize trust of actions that Github maintains. In order of preference, Hashicorp developed scripts or actions, followed by Github maintained, and trailed lastly by third party actions preserve best the platform's security. This allows to best maintain platform dependencies and ensure they're validated at all times.

🔗 External Links
Jira ticket detailing change to Github-supported actions: [TCE-2019](https://hashicorp.atlassian.net/browse/TCE-2019).

Cloud-ui PR: https://github.com/hashicorp/cloud-api/pull/152

Cloud-api PR: https://github.com/hashicorp/cloud-api/pull/152

### CODE REVIEW

File: .github/workflows/publish-sdk.yml [hunk 1]
```diff
@@ -58,22 +58,26 @@ jobs:
...
-
-            Changes:
-            - Pulled the latest public service SDK from ${{ github.event.inputs.service }}
-          delete-branch: true
+        run: |
+          cd $GITHUB_WORKSPACE/hcp-sdk-go
+          gh pr create --title "$PR_TITLE" --body "$PR_BODY" -H "$PR_SOURCE" -B "$PR_TARGET"
+        env:
+          PR_TITLE: "[auto] Update ${{ github.event.inputs.service }} specs"
+          PR_BODY: "This is an auto-generated PR created as part of the public SDK pipeline to update API specs for ${{ github.event.inputs.service }}."
```
Comment (bcmdarroch) [inline .github/workflows/publish-sdk.yml:77]:
```suggestion
          PR_BODY: "This is an auto-generated PR created as part of the public SDK pipeline to update the SDK for ${{ github.event.inputs.service }}."
```

**Comment by JolisaBrownHashiCorp** [comment]
🛠️ Description
These changes in the publish-api yaml file ensure that the pipeline's auto-generated branch for creating PRs is fresh upon user commit. By default, the auto-generated branch branch is deleted after every successful creation of a PR. However, when commits are made by the pipeline without resulting in a PR, the auto-generated branch remains in remote and may cause merge conflicts. This hinders the pipeline's successful creation of future PRs. 

📒 Additional Notes
This update is being executed in all services with write permissions for creating PRs. The three repos inclusive of these privileges are cloud-api, cloud-ui, and hcp-sdk-go as reflected in the PRs linked below.

❓ Why of the Changes
Allowing the pipeline to successfully create PRs reflecting updates to the API definition standardizes the automatic updates of grpc files, swaggers spec files, and OpenApi, helping ensure all changes are consistently aligned in our platform's SDKs.

🔗 External Links

Cloud-api PR: https://github.com/hashicorp/cloud-api/pull/160

Cloud-ui PR: https://github.com/hashicorp/cloud-ui/pull/2635

File: .github/workflows/publish-sdk.yml [hunk 2]
```diff
@@ -58,22 +58,28 @@ jobs:
           ./scripts/gen-go-service-sdk.sh $SERVICE
           rm -rf $GITHUB_WORKSPACE/hcp-sdk-go/temp
 
+      - name: Create branch with updated service specs
+        run: |
+          cd $GITHUB_WORKSPACE/hcp-sdk-go
+          git config user.name "GitHub Actions"
+          git config user.email "<>"
+          service_branch_exists="$(git ls-remote --heads origin auto-update-${{ github.event.inputs.service }}-specs)"
```
Comment (bcmdarroch) [inline .github/workflows/publish-sdk.yml:66]: repeat of [this comment](https://github.com/hashicorp/cloud-api/pull/160/files#r882801750), but let's create a branch name variable to ensure consistency.

File: .github/workflows/publish-sdk.yml [hunk 3]
```diff
@@ -58,22 +58,28 @@ jobs:
           ./scripts/gen-go-service-sdk.sh $SERVICE
           rm -rf $GITHUB_WORKSPACE/hcp-sdk-go/temp
 
+      - name: Create branch with updated service specs
+        run: |
+          cd $GITHUB_WORKSPACE/hcp-sdk-go
+          git config user.name "GitHub Actions"
```
Comment (bcmdarroch) [inline .github/workflows/publish-sdk.yml:64]:
```suggestion
          git config user.name "HashiCorp Cloud Services"
```

File: .github/workflows/publish-sdk.yml [hunk 4]
```diff
@@ -58,22 +58,28 @@ jobs:
...
+          git config user.email "<>"
```
Comment (bcmdarroch) [inline .github/workflows/publish-sdk.yml:65]:
```suggestion
          git config user.email "59096967+hashicorp-cloud@users.noreply.github.com"
```

**Review by bcmdarroch — APPROVED** [review]
While testing, I realized we don't need the Verify step at the end anymore. That was only needed because the original action didn't return any errors on failure. But our create branch and open PR steps will return errors when they fail.

**Pull request merged**